linode/manager
# M3-7080: Improve Cypress support for non-Prod environments

## Description 📝
This makes some modifications to Cypress API request authentication/configuration to account for the `REACT_APP_API_ROOT` environment variable. This will allow Cypress tests to run against non-Prod environments.

## Major Changes 🔄
- Override API request URL when `REACT_APP_API_ROOT` is defined during Cypress start-up and during test execution.
- Fixes test issues caused by fetching Linode regions using the Prod endpoint and attempting to use that data in tests against non-Prod environments
- Log additional test account information when `CY_TEST_USER_REPORT` is defined.

## How to test 🧪
* We can rely on the automated E2E tests to confirm that Cypress runs as expected via CI and that no regressions have been introduced there
* To confirm that the changes work as intended, set `REACT_APP_API_ROOT` to a non-Prod environment, and update `MANAGER_OAUTH` accordingly, then rebuild Cloud Manager and run `yarn cy:run`
  * The tests do not necessarily need to pass for now -- it's expected that differences in environments may cause some failures since we've only supported tests against Prod up to this point. However, we want to confirm that no failures are caused by Cypress configuration issues
